src/seedu/addressbook/data/person/Address/Address.java

```diff
@@ -0,0 +1,126 @@
+package seedu.addressbook.data.person.address;
+
+import javafx.geometry.Pos;
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's address in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ */
+public class Address {
+
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in the format: ";
+    public static final String ADDRESS_CONSTRAINTS = "a/BLOCK, STREET, UNIT, POSTAL CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = "(.*), (.*), (.*), (.*)";
+    public static final String ADDRESS_SPLIT_REGEX = ", ";
+
+    private static final int ADDRESS_BLOCK_INDEX = 0;
+    private static final int ADDRESS_STREET_INDEX = 1;
+    private static final int ADDRESS_UNIT_INDEX = 2;
+    private static final int ADDRESS_POSTAL_CODE_INDEX = 3;
+
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
+
+    private boolean isPrivate;
+
+    /**
+     * Validates given address.
+     *
+     * @throws IllegalValueException if given address string is invalid.
+     */
+    public Address(String address, boolean isPrivate) throws IllegalValueException {
+        String trimmedAddress = address.trim();
+        this.isPrivate = isPrivate;
+        if (!isValidAddress(trimmedAddress)) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS + ADDRESS_CONSTRAINTS);
+        }
+        block = null;
+        street = null;
+        unit = null;
+        postalCode = null;
+        
+        setAddressComponents(trimmedAddress);
+    }
+
+    /**
+     * Returns true if a given string is a valid person address.
+     */
+    public static boolean isValidAddress(String test) {
+        return test.matches(ADDRESS_VALIDATION_REGEX);
+    }
+
+
+    private void setAddressComponents(String address) {
+        String[] components = address.split(ADDRESS_SPLIT_REGEX);
+
+        this.block = extractBlock(components);
```

Comment:
Good attempt on implementing SLAP. However, consider the abstraction levels of an `ARRAY.split` operation and an `extractBlock` method call.